src/SAWScript/Crucible/MIR/Setup/Value.hs

```diff
@@ -84,22 +90,31 @@ type instance MS.Codebase MIR = CollectionState
 
 data MIRCrucibleContext =
   MIRCrucibleContext
-  { _mccRustModule      :: RustModule
-  , _mccBackend         :: SomeOnlineBackend
-  , _mccHandleAllocator :: HandleAllocator
+  { _mccRustModule           :: RustModule
+  , _mccBackend              :: SomeOnlineBackend
+  , _mccSimContext           :: SimContext (SAWCruciblePersonality Sym) Sym MIR
+  , _mccSymGlobalState       :: SymGlobalState Sym
+  , _mccStaticInitializerMap :: MirStaticInitializerMap
   }
 
 type instance MS.CrucibleContext MIR = MIRCrucibleContext
 
 type instance MS.Pointer' MIR sym = Some (MirPointer sym)
 
+-- | A 'MirStaticInitializerMap' maps the 'GlobalVar's of each top-level static
+-- value in a 'Mir.RustModule' to its initializer value (post-Crucible
+-- translation). See @Note [Translating MIR statics in SAW]@ in
+-- "SAWScript.Crucible.MIR.Builtins" for more details on how this map is
+-- created.
+type MirStaticInitializerMap = MapF GlobalVar (RegValue' Sym)
+
 -- | Unlike @LLVMPointsTo@ and @JVMPointsTo@, 'MirPointsTo' contains a /list/ of
 -- 'MS.SetupValue's on the right-hand side. This is due to how slices are
 -- represented in @crucible-mir-comp@, which stores the list of values
 -- referenced by the slice. The @mir_points_to@ command, on the other hand,
 -- always creates 'MirPointsTo' values with exactly one value in the list (see
 -- the @firstPointsToReferent@ function in "SAWScript.Crucible.MIR.Override").
-data MirPointsTo = MirPointsTo MS.ConditionMetadata MS.AllocIndex [MS.SetupValue MIR]
+data MirPointsTo = MirPointsTo MS.ConditionMetadata (MS.SetupValue MIR) [MS.SetupValue MIR]
```

Comment:
The first commit in this PR is all about this change, where we represent the reference value in a `MirPointsTo` statement with a `SetupValue` instead of an `AllocIndex`. This is required to support things like `mir_points_to (mir_static ...) ...`, as `mir_static` references do not have an `AllocIndex`. There is a bit of code churn that results from this change, but it's mostly straightforward. (I'll note one exception in another inline comment.)